src/index.ts

```diff
@@ -1,123 +1,40 @@
+      ISRACARD_6_DIGITS: string;
     }
   }
 }
 
-declare namespace window {
-  const bnhpApp: any;
-}
-
-async function login(userCode: string, password: string, page: puppeteer.Page) {
-  await page.waitFor('#inputSend');
-
-  await page.type('#userID', userCode);
-  await page.type('#userPassword', password);
-
-  page.click('#inputSend');
-
-  const answers = await inquirer.prompt([
-    {
-      type: 'input',
-      name: 'SMSPassword',
-      message: 'Enter the code you got in SMS:',
-    },
-  ]);
-
-  await page.type('#codeForOtp', answers.SMSPassword);
-
-  await Promise.all([
-    page.waitForNavigation(),
-    page.keyboard.press('Enter'),
-    page.click('#buttonNo'),
-  ]);
-}
-
 export async function init() {
-  const VIEWPORT_WIDTH = 1024;
-  const VIEWPORT_HEIGHT = 768;
-
-  const BASE_URL = 'https://biz2.bankhapoalim.co.il/authenticate/logon/main';
+  /* this initiates browser and returns every scraper as function */
+  /* each scraper opens its own page                              */
   const browser = await puppeteer.launch({ headless: true });
 
-  const page = await browser.newPage();
-
-  await page.setViewport({
-    width: VIEWPORT_WIDTH,
-    height: VIEWPORT_HEIGHT,
-  });
-
-  await page.goto(BASE_URL);
-  await login(process.env.USER_CODE, process.env.PASSWORD, page);
-
-  const result = await page.evaluate(() => {
-    return window.bnhpApp.restContext;
-  });
-  const apiSiteUrl = `https://biz2.bankhapoalim.co.il/${result.slice(1)}`;
-
-  const API_DATE_FORMAT = 'YYYYMMDD';
-  const defaultStartMoment = moment().subtract(1, 'years').add(1, 'day');
-  const startDateString = defaultStartMoment.format(API_DATE_FORMAT);
-  const endDateString = moment().format(API_DATE_FORMAT);
-  // TODO: https://www.npmjs.com/package/node-fetch-cookies
-
-  // TODO: Bring back validation as an option flag to each function
-  // const ajv = new Ajv({ verbose: true });
-  // // TODO: Validate asyncrhniously
-  // const valid = ajv.validate(accountDataSchemaFile, accountDataResult);
-  // console.log(valid);
-  // console.log(ajv.errors);
-
   return {
-    getAccountsData: () => {
-      const accountDataUrl = `${apiSiteUrl}/general/accounts`;
-      return fetchGetWithinPage<AccountDataSchema>(page, accountDataUrl);
-    },
-    getILSTransactions: (account: {
-      bankNumber: number;
-      branchNumber: number;
-      accountNumber: number;
-    }) => {
-      const fullAccountNumber = `${account.bankNumber}-${account.branchNumber}-${account.accountNumber}`;
-      const ILSCheckingTransactionsUrl = `${apiSiteUrl}/current-account/transactions?accountId=${fullAccountNumber}&numItemsPerPage=200&retrievalEndDate=${endDateString}&retrievalStartDate=${startDateString}&sortCode=1`;
-
-      return fetchPoalimXSRFWithinPage<ILSCheckingTransactionsDataSchema>(
-        page,
-        ILSCheckingTransactionsUrl,
-        '/current-account/transactions'
-      );
+    hapoalim: async (options?: scraper.hapoalimOptions) => {
```

Comment:
Is an object of parameters better?
Maybe just regular list of parameters?

Reply:
I thought about it.
If its separate parameters, you'd have to do every parameter. on an object you can relay on the default inputs

what do you think?

Reply:
That's also true for regular parameters:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters

Reply:
yep, but if there will are multiple options, and >1 of them have default values.

export class hapoalimOptions {
  validateSchema: boolean = false;
  isBusiness: boolean = true;
}

by your method, if I want to configure isBusiness, I'd have to configure validateSchema first, wouldn`t I?
on object parameter it is possible.